src/Microsoft.Data.Sqlite.Core/SqliteDataReader.cs

```diff
@@ -446,7 +446,26 @@ public override float GetFloat(int ordinal)
         /// <param name="ordinal">The zero-based column ordinal.</param>
         /// <returns>The value of the column.</returns>
         public override Guid GetGuid(int ordinal)
-            => new Guid(GetBlob(ordinal));
+        {
+            var sqliteType = GetSqliteType(ordinal);
+            switch (sqliteType)
+            {
+                case raw.SQLITE_TEXT:
+                    return new Guid(GetString(ordinal));
+                case raw.SQLITE_BLOB:
+                    var bytes = GetBlob(ordinal);
+                    if (bytes.Length == 16)
+                    {
+                        return new Guid(bytes);
+                    }
+                    else
+                    {
+                        return new Guid(System.Text.Encoding.UTF8.GetString(bytes, 0, bytes.Length));
+                    }
+                default:
+                    return new Guid(GetBlob(ordinal));
```

Comment:
Hmm... maybe just default to `GetString` if it's not a `BLOB`? Then again, neither exception is really useful...
* *Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).*
* *Byte array for GUID must be exactly 16 bytes long.*

**But**, the `string` overload *could* be if it told you the value. E.g. "The value '3.14' is invalid. Guid should contain..."